src/realm/query_engine.hpp

```diff
@@ -1655,7 +1657,7 @@ template <class TConditionFunction>
 class StringNode : public StringNodeBase {
 public:
     constexpr static bool case_sensitive_comparison =
-        is_any_v<TConditionFunction, Greater, GreaterEqual, Less, LessEqual>;
+        is_any_v<TConditionFunction, NotEqual, Greater, GreaterEqual, Less, LessEqual>;
```

Comment:
Why adding `NotEqual` to case-insensitive comparison?

Reply:
Because it was missing. Aparently we did not have any tests for that. (And it is case sentitive, not insensitive)

Reply:
I will try to make a failing test

Reply:
OK - it works anyway. But in many cases we construct the m_ucase and m_lcase when they are not needed.